src/test/test-errno-util.c

```diff
@@ -89,4 +89,18 @@ TEST(RET_GATHER) {
         assert_se(y == 3);
 }
 
+TEST(ERRNO_IS_TRANSIENT) {
+        assert_se( ERRNO_IS_NEG_TRANSIENT(-EINTR));
+        assert_se(!ERRNO_IS_NEG_TRANSIENT(EINTR));
+        assert_se( ERRNO_IS_TRANSIENT(-EINTR));
+        assert_se( ERRNO_IS_TRANSIENT(EINTR));
+
+        /* Test with type wider than int */
+        ssize_t r = -EAGAIN;
+        assert_se( ERRNO_IS_NEG_TRANSIENT(r));
+
+        r += 2 * (ssize_t) INT_MAX;
+        assert_se(!ERRNO_IS_NEG_TRANSIENT(r));
```

Comment:
On 32-bit architectures where sizeof(ssize_t) == sizeof(int) this would result to signed integer overflow.
If you need to rely on the result of integer overflow, please use an unsigned integer type.

Reply:
Duh, of course.